src/pt/kuromangas/src/eu/kanade/tachiyomi/extension/pt/kuromangas/KuroMangas.kt

```diff
@@ -0,0 +1,277 @@
+package eu.kanade.tachiyomi.extension.pt.kuromangas
+
+import androidx.preference.EditTextPreference
+import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.lib.cookieinterceptor.CookieInterceptor
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import keiyoushi.utils.getPreferencesLazy
+import keiyoushi.utils.parseAs
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class KuroMangas : HttpSource(), ConfigurableSource {
+
+    override val name = "KuroMangas"
+
+    override val baseUrl = "https://beta.kuromangas.com"
+
+    override val lang = "pt-BR"
+
+    override val supportsLatest = true
+
+    private val preferences by getPreferencesLazy()
+
+    private val apiUrl = "$baseUrl/api"
+
+    private val cdnUrl = "https://cdn.kuromangas.com"
+
+    override val client by lazy {
+        val token = getToken()
+        network.cloudflareClient.newBuilder()
+            .rateLimit(2)
+            .apply {
+                if (token.isNotEmpty()) {
+                    addInterceptor(CookieInterceptor(API_HOST, "token" to token))
+                    addInterceptor { chain ->
+                        val request = chain.request().newBuilder()
+                            .header("Authorization", "Bearer $token")
+                            .build()
+                        chain.proceed(request)
+                    }
+                }
+            }
+            .build()
+    }
+
+    override fun headersBuilder(): Headers.Builder = super.headersBuilder()
+        .add("Accept", "application/json, text/plain, */*")
+        .add("Accept-Language", "pt-BR,pt;q=0.8,en-US;q=0.5,en;q=0.3")
+        .add("Referer", baseUrl)
+
+    private val dateFormat by lazy {
+        SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.ROOT)
```

Comment:
Don't use a literal `Z`, it will use your local timezone.
```suggestion
        SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ", Locale.ROOT)
```
otherwise you have to set the timezone explicitly:
```suggestion
        SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.ROOT).apply {
            timeZone = TimeZone.getTimeZone("UTC")
        }
```

(this is assuming the timestamp is _actually_ shown in UTC)

Reply:
timestamp come in iso 8601
```
"upload_date": "2025-12-12T20:19:08.188Z"
"created_at": "2025-12-12T19:34:46.030Z"
```

so -> 
```
private val dateFormat by lazy {
        SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.ROOT).apply {
            timeZone = TimeZone.getTimeZone("UTC")
        }
```